cpg_workflows/large_cohort/site_only_vcf.py

```diff
@@ -55,27 +221,62 @@ def vds_to_site_only_ht(
     if can_reuse(out_ht_path):
         return hl.read_table(str(out_ht_path))
 
-    mt = hl.vds.to_dense_mt(vds)
+    # Passing un-densified MatrixTable to default_compute_info
+    mt = vds.variant_data
 
-    mt = mt.filter_cols(hl.len(sample_qc_ht[mt.col_key].filters) > 0, keep=False)
-    mt = mt.filter_cols(hl.is_defined(relateds_to_drop_ht[mt.col_key]), keep=False)
+    # Compute and checkpoint the allele specific info annotations after recomputing
+    # AS_QUALapprox from LPL, and fixing the length of AS_SB_TABLE, AS_RAW_MQ,
+    # AS_RAW_ReadPosRankSum and AS_RAW_MQRankSum.
+    mt = mt.annotate_rows(alt_alleles_range_array=hl.range(1, hl.len(mt.alleles)))
+    correct_mt = mt.annotate_entries(
+        gvcf_info=mt.gvcf_info.annotate(
+            AS_QUALapprox=recompute_as_qualapprox_from_lpl(mt),
+            **correct_as_annotations(mt),
+        ),
+    )
 
-    mt = _filter_rows_and_add_tags(mt)
-    var_ht = _create_info_ht(mt, n_partitions=mt.n_partitions())
-    var_ht = var_ht.checkpoint(out_ht_pre_vcf_adjusted_path, overwrite=True)
-    var_ht = adjust_vcf_incompatible_types(
-        var_ht,
-        # with default INFO_VCF_AS_PIPE_DELIMITED_FIELDS, AS_VarDP will be converted
-        # into a pipe-delimited value e.g.: VarDP=|132.1|140.2
-        # which breaks VQSR parser (it doesn't recognise the delimiter and treats
-        # it as an array with a single string value "|132.1|140.2", leading to
-        # an IndexOutOfBound exception when trying to access value for second allele)
-        pipe_delimited_annotations=[],
+    correct_mt = correct_mt.filter_cols(hl.len(sample_qc_ht[correct_mt.col_key].filters) > 0, keep=False)
+    correct_mt = correct_mt.filter_cols(hl.is_defined(relateds_to_drop_ht[correct_mt.col_key]), keep=False)
```

Comment:
We should drop the samples we intend to remove first, so they aren't included in the AS_QUALapprox recomputation.